custom_components/eldes_alarm/sensor.py

```diff
@@ -111,3 +113,36 @@ def icon(self):
     def native_value(self):
         """Return the state of the sensor."""
         return self.data["info"].get("phoneNumber", "")
+
+
+class EldesTemperatureSensor(EldesDeviceEntity, SensorEntity):
+    """Class for the temperature sensor."""
+
+    @property
+    def unique_id(self):
+        """Return a unique identifier for this entity."""
+        return f"{self.imei}_{self.__get_temp()['sensorName']}_{self.__get_temp()['sensorId']}_temperature"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self.__get_temp()['sensorName']} temperature"
+
+    @property
+    def icon(self):
```

Comment:
I think instead of `icon`, you can add `state_class` as `temperature`. And then icon will be automatically assigned, also `native_unit_of_measurement` is also optional after you provide `state_class`. Try to test these changes. Here is sensor entity documentation https://developers.home-assistant.io/docs/core/entity/sensor/